### PR TITLE
Folders Frontend (4) 

### PR DESCRIPTION
**Pull request by g-luo**
* Add modal to move document folder by creating move folder event in docs-component upon list item click. 
* If document is created while a folder is open, the document is automatically added to that folder.
* Cache all documents in the frontend and filter by folderID when rendering in docs-component to reduce lag time between servlet request and response

### CODE REVIEW

**Review by tin0d0 — APPROVED** [review]
lgtm!

**Review by max-friedman — APPROVED** [review]
lgtm!

**Review by jileiwang-google — APPROVED** [review]
LGTM

File: project/src/main/java/com/google/sps/Servlets/UserHomeServlet.java [hunk 1]
```diff
@@ -47,10 +47,20 @@ private String convertToJson(Object obj) {
...
     String language = request.getParameter("language");
-    String documentID = request.getParameter("documentID");
+    String docHash = request.getParameter("docHash");
     long userID = (long) request.getSession(false).getAttribute("userID");
-    // check if null, then redirect
-    Database.createDocument(name, language, documentID, userID);
-    response.sendRedirect("/Document?documentHash=" + documentID);
+    Database.createDocument(name, language, docHash, userID);
+    String folderID = request.getParameter("folderID");
+    if (validFolderID(folderID)) {
```
Comment (HidyHan) [inline project/src/main/java/com/google/sps/Servlets/UserHomeServlet.java]: Naming nit: IsValidFolderID

**Review by HidyHan — APPROVED** [review]

**Review by frankpzn — APPROVED** [review]

**Pull request merged**